goalc/compiler/compilation/Asm.cpp

```diff
@@ -399,7 +605,56 @@ Val* Compiler::compile_asm_vf_math3(const goos::Object& form,
         src2->print());
   }
 
-  env->emit_ir<IR_VFMath3Asm>(color, dest, src1, src2, kind);
+  u8 mask = 0b1111;
+  if (args.has_named("mask")) {
+    try {
+      mask = args.named.at("mask").as_int();
+    } catch (std::exception) {
+      throw_compiler_error(
+          form,
+          "The value {} is invalid for a destination mask, it could not be evaluated as a "
+          "constant integer.",
+          args.named.at("mask").print());
+    }
+
+    if (mask < 0 || mask > 15) {
+      throw_compiler_error(form, "The value {} is out of range for a blend mask (0-15 inclusive).",
+                           mask);
+    }
+  }
+
+  // If there is a broadcast register, splat that float across the entire src2 register before
+  // performing the operation For example vaddx.xyzw vf10, vf20, vf30
+  // vf10[x] = vf20[x] + vf30[x]
+  // vf10[y] = vf20[y] + vf30[x]
+  // vf10[z] = vf20[z] + vf30[x]
+  // vf10[w] = vf20[w] + vf30[x]
+  // TODO - surely i can condense this...(only using a temp register when i have to)
```

Comment:
I don't think there's much more you can do here to make it simpler.

If it's possible for `temp_reg` to be the same register as `src2`, the compiler will do this automatically during the register allocation phase.